exporter.go

```diff
@@ -106,23 +141,40 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 	mapper.mappingsCountMetric.Collect(ch)
 }
 
+func (e *Exporter) retryToDialBeanstalk() (*beanstalk.Conn, error) {
+	for i := 0; i < e.config.numberOfRetry; i++ {
+		c, err := beanstalk.Dial("tcp", e.config.address)
+		if err == nil {
+			return c, nil
+		}
+		e.scrapeConnectionErrorMetric.Inc()
+		log.Errorf("Error. Can't connect to beanstalk: %v retrying", err)
+		time.Sleep(e.config.retrySleepTimer)
+	}
+	return nil, fmt.Errorf(" %d connection retry was failed. skipping current stats", e.config.numberOfRetry)
+}
+
 // scrape retrieves all the available metrics and invoke the given callback on each of them.
 func (e *Exporter) scrape(f func(prometheus.Collector)) {
 	start := time.Now()
 	defer func() {
 		e.scrapeHistogramMetric.Observe(time.Since(start).Seconds())
 	}()
-
 	// system stats
-	c, err := beanstalk.Dial("tcp", e.address)
+	c, err := beanstalk.Dial("tcp", e.config.address)
 	if err != nil {
 		e.scrapeConnectionErrorMetric.Inc()
-		log.Fatalf("Error. Can't connect to beanstalk: %v", err)
-		return
+		log.Errorf("Error. Can't connect to beanstalk: %v", err)
+		c, err = e.retryToDialBeanstalk()
+		if err != nil {
+			log.Errorf("Error. can't connect to beanstalk: %v", err)
```

Comment:
Shouldn't this be fatal? 
```suggestion
			log.Fatalf("Error. can't connect to beanstalk: %v", err)
```

As we don't want it to hang forever, and we have alerts now that report when the exporter is down, while we don't have alerts if it's stuck.

Reply:
I think alerts should check `scrapeConnectionErrorMetric` metric instead of going down.

Reply:
Agree, but then this can not be deployed until those alerts are set.